File: Webapp/Front-end/src/app/endpoints/popust-endpoints/popust-getall.endpoint.ts
import {Injectable} from "@angular/core";
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {MojConfig} from "../../moj-config";

@Injectable({providedIn: 'root'})
export class PopustGetAllEndpoint implements MyBaseEndpoint<void, PopustGetAllResponse>
{
  constructor(public httpClient: HttpClient) {
  }
  obradi(request:void) :Observable<PopustGetAllResponse>
  {
    let url=MojConfig.adresa_servera+'/popust/get-all';
    return this.httpClient.get<PopustGetAllResponse>(url);
  }
}
export interface PopustGetAllResponse {
  popusti: PopustGetAllResponsePopusti[]
}

export interface PopustGetAllResponsePopusti {
  id: number
  naziv: string
  datumDo: string
  datumOd: string
  procenat: number
  isDeleted: boolean
}

File: Webapp/Api/PC_Web_Shop/Endpoints/PopustEndpoints/PopustGetAll/PopustGetAllEndpoint.cs

using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using PC_Web_Shop.Data;
using PC_Web_Shop.Helper;

namespace PC_Web_Shop.Endpoints.PopustEndpoints.PopustGetAll
{
    [Route("popust")]
    public class PopustGetAllEndpoint:MyBaseEndpoint<PopustGetAllRequest,PopustGetAllResponse>
    {
        private readonly ApplicationDbContext _applicationDbContext;

        public PopustGetAllEndpoint(ApplicationDbContext applicationDbContext)
        {
            _applicationDbContext = applicationDbContext;
        }

        [HttpGet("get-all")]

        public override async Task<PopustGetAllResponse> Obradi([FromQuery] PopustGetAllRequest request, CancellationToken cancellationToken) {

            var popust = await _applicationDbContext.Popust.Where(x => x.IsDeleted == false)
                .Select(x=> new PopustGetAllResponsePopust
                {
                Id=x.Id,
                Naziv=x.Naziv,
                DatumOd=x.DatumOd,
                DatumDo=x.DatumDo,
                Procenat=x.Procenat,
                IsDeleted=x.IsDeleted
                }
                ).ToListAsync(cancellationToken);
            return new PopustGetAllResponse
            {
                Popusti = popust
            };
        }
    }
}
